src/main/java/com/minecolonies/coremod/client/render/worldevent/HighlightManager.java

```diff
@@ -1,129 +1,128 @@
 package com.minecolonies.coremod.client.render.worldevent;
 
-import com.ldtteam.structurize.util.WorldRenderMacros;
-import net.minecraft.client.Minecraft;
-import net.minecraft.client.renderer.MultiBufferSource;
-import net.minecraft.core.BlockPos;
-import net.minecraft.world.phys.AABB;
+import com.minecolonies.coremod.client.render.worldevent.highlightmanager.IHighlightRenderData;
 
-import java.util.*;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
 
 public class HighlightManager
 {
     /**
      * A position to highlight with a unique id.
      */
-    private static final Map<String, List<TimedBoxRenderData>> HIGHLIGHT_MAP = new HashMap<>();
+    private static final List<HighlightRenderDataContainer> HIGHLIGHT_ITEMS = new ArrayList<>();
```

Comment:
That was a map on purpose to avoid re-inserting the same element again. Either we make this a set, and make a unique hash per obj, or should keep it a map.